### PR TITLE
UCS/CALLBACKQ: API to remove elements according to arbitrary condition

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5664/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5665/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5676/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5688/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5697/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@alinask pls take a look

File: src/ucs/datastruct/callbackq.h [hunk 1]
```diff
@@ -142,14 +165,30 @@ int ucs_callbackq_add_safe(ucs_callbackq_t *cbq, ucs_callback_t cb, void *arg,
  * be removed at some point in the near future.
  * This can be used from any context and any thread, including but not limited to:
  * - A callback can remove another callback or itself.
- * - A thread remove add a callback while another thread is dispatching callbacks.
+ * - A thread remove a callback while another thread is dispatching callbacks.
```
Comment (alinask) [inline src/ucs/datastruct/callbackq.h]: A thread _can't_ remove... ?

**Pull request merged**